brayns/network/entrypoints/ModelMaterialEntrypoint.h

```diff
@@ -134,83 +141,89 @@ class GetMaterialEntrypoint : public Entrypoint<GetModelMessage, MaterialType>
 public:
     using Request = typename Entrypoint<GetModelMessage, MaterialType>::Request;
 
-    GetMaterialEntrypoint(Scene &scene)
-        : _scene(scene)
+    GetMaterialEntrypoint(ModelManager &models)
+        : _models(models)
     {
     }
 
     virtual void onRequest(const Request &request) override
     {
         auto params = request.getParams();
         auto modelId = params.id;
-        auto &instance = _scene.getModelInstance(modelId);
+        auto &instance = ExtractModel::fromId(_models, modelId);
         Model &model = instance.getModel();
-        auto &component = model.getComponent<MaterialComponent>();
-        auto &material = component.getMaterial();
-        if (auto cast = material.as<MaterialType>())
+        auto &components = model.getComponents();
+
+        auto material = components.find<Material>();
+        if (!material)
+        {
+            throw JsonRpcException("The model does not have material");
+        }
+
+        if (auto cast = material->as<MaterialType>())
```

Comment:
Just as "common practice", I would do all the checks before doing the actual work (even if it takes one more line) but you can leave it like that.

Reply:
Addressed